dir.props

```diff
@@ -117,6 +117,27 @@
   </PropertyGroup>
 
   <Import Project="$(BuildConfigurationImportFile)" Condition="Exists('$(BuildConfigurationImportFile)')" />
+  <PropertyGroup>
+    <_runtimeOSVersionIndex>$(RuntimeOS.IndexOfAny(".-0123456789"))</_runtimeOSVersionIndex>
+    <_runtimeOSFamily Condition="'$(_runtimeOSVersionIndex)' != '-1'">$(RuntimeOS.SubString(0, $(_runtimeOSVersionIndex)))</_runtimeOSFamily>
+    <_portableRID>linux</_portableRID>
+    <_portableRID Condition="'$(_runtimeOSFamily)' == 'win'">win</_portableRID>
+    <_portableRID Condition="'$(_runtimeOSFamily)' == 'osx'">osx</_portableRID>
+    <!-- support cross-targeting by choosing a RID to restore when running on a different machine that what we're build for -->
+    <_portableRID Condition="'$(OSGroup)' == 'Unix' AND '$(_runtimeOSFamily)' != 'osx'">linux</_portableRID>
```

Comment:
I'm not sure how cross-targeting works and if/how it relates to portable builds. Can you give an example of how to cross-target on a Linux machine? I can move it back to runtime.depproj but then it will no longer affect the package name.